runtimes/data-binding.mdx

```diff
@@ -0,0 +1,169 @@
+
+        // Create by index
+        for (i in 0 until vm.instanceCount) {
+            val vmiIndexed = vm.createInstanceFromIndex(i)
+        }
+
+        // Create by name
+        val vmiNamed = vm.createInstanceFromName("My Instance")
+        ```
+    </Tab>
+</Tabs>
+
+# View Model Instances
+
+Once you've used your view model to create an instance, that instance can be assigned to an artboard. This establishes the bindings set up at edit time.
+
+<Note>The initial values of the instance are not applied to their bound elements until the artboard is advanced.</Note>
+
+<Tabs>
+    <Tab title="Android">
+        ```kotlin
+        view.setRiveResource(
+            R.raw.my_rive_file,
+            artboardName = "My Artboard",
+        )
+
+        val vm = view.controller.file?.getViewModelByName("My View Model")!!
+        val vmi = vm.createInstanceFromName("My Instance")
+
+        // Apply the instance to the artboard
+        view.controller.activeArtboard?.setViewModel(vmi)
+        ```
+    </Tab>
+</Tabs>
+
+# Properties
+
+Properties descriptors can be inspected on a view model to discover at runtime which are available. These are not the mutable properties themselves though - once again those are on instances. These descriptors have a type and name.
+
+<Tabs>
+    <Tab title="Android">
+        ```kotlin
+        val vm = view.controller.file?.getViewModelByName("My View Model")!!
+
+        // A list of properties
+        val properties = vm.properties
+        assertContains(
+            properties,
+            ViewModel.Property(ViewModel.PropertyDataType.NUMBER, "My Number Property")
+        )
+        ```
+    </Tab>
+</Tabs>
+
+View model instances have mutable properties. References to these properties can be retrieved by name. They have get, set, and observe operations. Getting or observing the value will retrieve the latest value set on that properties binding, as of the last artboard advance. Setting the value will update the value and all of its bound elements. Note that trigger properties can only be set.
+
+<Note>After setting a property's value, the changes will not apply to their bound elements until the artboard is advanced.</Note>
+
+<Tabs>
+    <Tab title="Android">
+        ```kotlin
+        val vm = view.controller.file?.getViewModelByName("My View Model")!!
+        val vmi = vm.createInstanceFromName("My Instance")
+
+        val numberProperty = vmi.getNumberProperty("My Number Property")
+        // Get
+        val myPropertyValue = numberProperty.value
+        // Set
+        myProperty.value = 10f
+        ```
+    </Tab>
+</Tabs>
+
+# Enums
+
+Enums properties come in two flavors: system and user-defined. In practice, you will not need to worry about the distinction, but just be aware that system enums are always available in all Rive files, representing options from the editor's dropdowns, where user-defined enums are those defined by a designer in the editor.
```

Comment:
@bodymovin to clarify, but as I understood it the system enums won't always be on every file, but it be populated in a file if that file uses system enums